### ISSUE TITLE
Production servers are hitting Stack Overflow access quota

### ISSUE THREAD

**Issue by kunaltyagi** [badges/shields#3591]
Are you experiencing an issue with...

- [x] [shields.io](https://shields.io/#/)
- [ ] My own instance
- [ ] [gh-badges NPM package](https://www.npmjs.com/package/gh-badges)

:beetle: **Description**

Badge for StackOverflow fails to provide the correct questions intermittently

:link: **Link to the badge**

https://img.shields.io/stackexchange/stackoverflow/t/point-cloud-library.svg?logo=stackoverflow

Live image:
![SO badge](https://img.shields.io/stackexchange/stackoverflow/t/point-cloud-library.svg?logo=stackoverflow)

Discussion: https://github.com/PointCloudLibrary/pcl/issues/3175

:bulb: **Possible Solution**
This might not be the cause, but caching SO results might prevent a non-response from their end (if that's the cause).

Manually calling the SO API that shields.io is supposed to call results in a correct response which led to this hypothesis.

**Comment by paulmelnikow** [comment]
Hi, thanks for the report!

This is working fine for me locally:

```
~/c/shields (master|✔) $ npm run badge "https://img.shields.io/stackexchange/stackoverflow/t/point-cloud-library.svg?logo=stackoverflow"

> shields.io@0.0.0 badge /Users/pnm/code/shields
> cross-env NODE_CONFIG_ENV=test TRACE_SERVICES=true node scripts/badge-cli.js "https://img.shields.io/stackexchange/stackoverflow/t/point-cloud-library.svg?logo=stackoverflow"

0620125711 FsTokenPersistence configured with /Users/pnm/code/shields/private/github-user-tokens.json
0620125711 Server is starting up: http://localhost:1111/
 Service class  👩‍🍳
 StackExchangeQuestions
 Named params  🎫
 { stackexchangesite: 'stackoverflow',
  query: 'point-cloud-library' }
 Query params  🖍
 { color: undefined,
  colorA: undefined,
  colorB: undefined,
  label: undefined,
  labelColor: undefined,
  link: undefined,
  logo: 'stackoverflow',
  logoColor: undefined,
  logoPosition: undefined,
  logoWidth: undefined,
  style: undefined }
 Request  🏹
 https://api.stackexchange.com/2.2/tags/point-cloud-library/info
 Response status code  🎯
 200
 Response JSON (before validation)  🎯
{ items:
   [ { has_synonyms: false,
       is_moderator_only: false,
       is_required: false,
       count: 1035,
       name: 'point-cloud-library' } ],
  has_more: false,
  quota_max: 300,
  quota_remaining: 299 }
 Response after validation  🛁
{ items: [ { count: 1035 } ] }
 Service data  🛡
 { label: 'stackoverflow point-cloud-library questions',
  message: '1k',
  color: 'yellowgreen' }
 Rendered badge  🛡
 { label: 'stackoverflow point-cloud-library questions',
  message: '1k',
  color: 'yellowgreen',
  link: [],
  name: 'stackoverflow point-cloud-library questions',
  value: '1k' }
```

I noticed this:

```
  quota_max: 300,
  quota_remaining: 299 }
```

Could we be hitting an API usage limit in production?

**Comment by kunaltyagi** [comment]
> Could we be hitting an API usage limit in production?

This seems most likely as I noted above. [SO API docs](https://api.stackexchange.com/docs/throttle) also mention the throttling and quota restrictions.

**Comment by PyvesB** [comment]
I found [this question/answer](https://stackapps.com/questions/67/how-api-keys-work) more helpful.  It describes the 300 quota max mentioned by @paulmelnikow. It's a daily limit per IP, which seems very low, I'm not surprised we're hitting the limit on our production servers.

I managed to generate an API key quite easily, and indeed the max quota indicated in the response was increased to 10000. I suggest we add support for an optional API token like we have in other services and put a key on the production servers.
